lib/utils.js

```diff
@@ -161,6 +161,40 @@ NodeArgInfo.prototype.isImportant = function () {
   return false
 }
 
+/**
+ * Class for providing and storing unique ids for objects.
+ * @constructor
+ */
+function Oid() {
+  /**
+   * Map of object to its identifier. We use WeakMap to avoid keeping references to the objects
+   * and potentially leaking memory.
+   * @type {Object}
+   */
+  this._objectToId = new WeakMap()
```

Comment:
Unfortunately there is a type error here:

```
/Users/heatherarthur/src/shepherd/lib/utils.js:174: ERROR - variable WeakMap is undeclared
  this._objectToId = new WeakMap()
```

Reply:
haha this still uses closure externs for type-checking?! :older_man: 

https://github.com/Medium/shepherd/tree/master/externs
https://github.com/google/closure-compiler/blob/50713dceb16b369f8f317e5b49e8a9dec2b47496/externs/es6_collections.js#L112

Reply:
Do those links that mean I can do something easy to fix it, or not really?

Reply:
ya you can just throw a WeakMap definition in the externs dir in shepherd